### ISSUE TITLE
Redesign GMT splash screen

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#1276]
Currently, the GMT splash screen is
```

	GMT - The Generic Mapping Tools, Version 6.0.0rc3 [64-bit] [8 cores]
	(c) 1991-2019 The GMT Team (https://www.generic-mapping-tools.org/team.html).

Supported in part by the US National Science Foundation (http://www.nsf.gov/)
and volunteers from around the world.

GMT is distributed under the GNU LGP License (http://www.gnu.org/licenses/lgpl.html).

usage: gmt [options]
       gmt <module name> [<module-options>]

session management:
  gmt begin         Initiate a new GMT session using modern mode [classic].
  gmt end           Terminate the current GMT modern mode session.
  gmt docs          Display the HTML documentation for selected module.
  gmt figure        Set figure format specifics under a GMT modern mode session.
  gmt inset         Manage figure inset setup and completion.
  gmt subplot       Initiate a multi-panel figure.
  gmt clear         Delete gmt.history, gmt.conf, current CPT, user cache or data, or all.

options:
  --help            List descriptions of available GMT modules.
  --show-bindir     Show directory with GMT executables.
  --show-cores      Show number of available cores.
  --show-datadir    Show directory/ies with user data.
  --show-modules    Show all module names.
  --show-library    Show path of the shared GMT library.
  --show-plugindir  Show directory for plug-ins.
  --show-sharedir   Show directory for shared GMT resources.
  --version         Print GMT version number.

if <module-options> is '=' we call exit (0) if module exist and non-zero otherwise.
```

Some possible improvements:

- [x] Remove the session management part. 
- [x] list session management  "modules" in `gmt --help`
- [x] list session management  "modules" `gmt --show-modules` (maybe not necessary)
- [x] Add `gmt --show-citation` to print the latest GMT paper?
- [ ] Add URLs to the GMT main site, documentation, issue page and forum

**Comment by seisman** [comment]
What's the reason not listing these modules in gmt --help and gmt --show-modules?
https://github.com/GenericMappingTools/gmt/blob/38d9c79529cb6fc42f7773ab484314a744ea21c1/src/gmt_core_module.c#L243-L254

**Comment by PaulWessel** [comment]
OK, here is my take on all this and why I made it the way it is:  GMT modern mode is composed of

1. Simplified syntax (the -**O -K -P, -R -J** stuff)
2. New modules movie and events, plus renamed modules without ps
3. Commands that gives structure to the GMT session

Despite the brief discussion we had in La Jolla, I still see begin, end, figure, inset, subplot, clear, and even docs as completely different beasts than the rest.  None of them actually make any plotting, they just structure things.  All plotting is done by the classic modules and all data analysis are done by the classic module.  These commands are more part of a very terse GMT "language" than the "modules" of that language.  They change the state of the session, unlike any of the modules.

If you agree with this view of the world then it makes sense to not group the commands with the modules.  However, we can have a discussion with all in @GenericMappingTools/core about this, now.

**Comment by seisman** [comment]
I agree that begin et al. are different from other modules, but at least we should list them at some place. Maybe list them in `gmt --help`, but not in `gmt --show-modules`?

**Comment by PaulWessel** [comment]
Picking up on this.  gmt --help shows classic mode names only - do we want to switch to modern mode names?

**Comment by seisman** [comment]
Yes, I prefer the modern names.

**Comment by PaulWessel** [comment]
May have to wait.  While changing the output names is doable, changing the order to be alphabetically correct for modern will take more work.

**Comment by PaulWessel** [comment]
We could try.  Would you have a quick python/script solution to

1. Replace all instances in *.c *.sh of THIS_MODULE_NAME with THIS_MODULE_CLASSIC_NAME
2. Add the line #define THIS_MODULE_MODERN_NAME after the #define THIS_MODULE_CLASSIC_NAME line and assign the classic name except remove any leading ps
3. Fix the special cases colorbar, plot, and plot3d manually.

then I can probably change the gmt_make_module_src.sh to produce more information in gmt_*_module.[ch] so that we can report modern names sorted correctly. Build and test after step 1 just to be sure.

**Comment by PaulWessel** [comment]
Hi @seisman, are you able to help with this? i would be good enough to just duplicate the CLASSIC name onto the new MODERN line and I can manually change the ps and special cases. I can probably do this myself with an awk statement though, so if you have not started let me know and I will have a look.

**Comment by seisman** [comment]
@PaulWessel @joa-quim @leouieda I think we can close the issue if we're not going to add the links (homepage, documentation, and forum) to the splash screen.

**Comment by PaulWessel** [comment]
Yes, I think so.  But maybe we could add a new target to gmt docs, called "home". so

`gmt docs home
`

will open www.generic-mapping-tools.org.  Could be some other name though, like "website" or maybe even nothing? Right now,

`gmt docs
`

gives

> docs [ERROR]: No files or documents given

**Comment by seisman** [comment]
`gmt docs` prints module usage for me.

Anyway, I think we should open a new issue to discuss if more targets are needed for `gmt docs`.

**Comment by PaulWessel** [comment]
OK, that is fine.  I will try to figure out why `gmt docs` gives me an error.
